src/op_sync_team.rs

```diff
@@ -268,13 +282,15 @@ pub fn sync_team(
                 noops.inc();
                 trace!(logger, "Keeping user on the team");
             }
-            TeamAction::Remove(github_name) => {
+            TeamAction::Remove(github_name, github_id) => {
                 let logger = logger.new(o!(
                     "github-name" => format!("{}", github_name),                ));
 
                 removals.inc();
                 info!(logger, "Removing user from the team");
                 if do_it_live {
+                    invited.remove(&github_id);
```

Comment:
We should only remove them from the `invited` list if they are successfully removed (down below, around line 320 with the current changes, could be changed into a `match` on the result of `rt.block_on(...)`).

Reply:
Doh, of course :)